### DISCUSSION TITLE
UI always shows UTC

### DISCUSSION THREAD

**Discussion by ascl00** [blakeblackshear/frigate#13997] [General]
I'm failing to get any part of the UI to show local time, it's always showing UTC.

The (baremetal, debian) host this is running on shows:

```
root@frigate:~# timedatectl 
               Local time: Fri 2024-09-27 15:21:24 AEST
           Universal time: Fri 2024-09-27 05:21:24 UTC
                 RTC time: Fri 2024-09-27 05:21:24
                Time zone: Australia/Sydney (AEST, +1000)
System clock synchronized: yes
              NTP service: n/a
          RTC in local TZ: no
root@frigate:~# 
```

I have the timezone explicitly set in the config:
```
ui:
  timezone: Australia/Sydney
  date_style: full
```
which is reflected when the config is queried via HTTP:
```
# wget -q -O - 'http://192.168.0.19:5000/api/config' | json_pp | grep timezone
      "timezone" : "Australia/Sydney"
```

My docker-compose.yml shows:
```    volumes:
      - /etc/localtime:/etc/localtime:ro
```
and localtime is linked correctly:
```
$ ls -la /etc/localtime 
lrwxrwxrwx 1 root root 36 Sep 24 20:14 /etc/localtime -> /usr/share/zoneinfo/Australia/Sydney
```
Every UI element shows UTC (including event stream etc)
![image](https://github.com/user-attachments/assets/4d8fc561-be46-4685-970c-0d840c5be158)

Logs however, show as localtime:
![image](https://github.com/user-attachments/assets/394e43b6-837a-4e24-bc8f-5541269c4555)

Weirdly I do see a single nginx log entry that includes the timezone:
``` 
[27/Sep/2024:15:08:23 +1000] "GET /api/review/summary?timezone=Australia%2FSydney
```

But I guess this comes from the config.

Browser is correctly set (googling "local time" gets the right time from google.com for eg).

What am I doing wrong here?

**Comment by NickM-27** [comment]
the times shown on the graph for example come from the browser, so it seems the browser is set to UTC or there is some privacy setting not allow the site to know what the timezone is configured as (this is common in firefox).

**Comment by ascl00** [comment]
You are absolutely correct! if I use hit the URL in edge, I get correct times! Excellent, thank you (now to figure out which FF setting it is...).
 
That said, isn't the ui.timezone option supposed to override the browser setting? Should I create a bug for this?

**Reply by NickM-27** [reply]
The browser can't override the timezone if it isn't allowed to know what the browsers timezone is

**Comment by ascl00** [comment]
For reference for anyone reading this later: https://support.mozilla.org/en-US/questions/1297208

Thanks Nick, much appreciated, I seem to be doing a good job of hitting all the gotchas! :)
